sdk/storage/azdatalake/file/models.go

```diff
@@ -61,6 +59,18 @@ type CreateOptions struct {
 	ACL *string
 }
 
+// CreateExpiryValues describes when a newly created file should expire.
+// A zero-value indicates the file has no expiration date.
+type CreateExpiryValues struct {
+	// ExpiresOn contains the time the file should expire.
```

Comment:
I would just reverse the order of the fields in the struct because the mental model is:
1. Select an expiry type first and then
2. Set ExpiresOn (if necessary)